File: src/com/sierrawireless/avphone/task/LogoutTask.java
package com.sierrawireless.avphone.task;

import java.io.IOException;

import net.airvantage.model.AirVantageException;
import net.airvantage.model.AvError;
import net.airvantage.utils.IAirVantageClient;
import android.os.AsyncTask;

public class LogoutTask extends AsyncTask<String, Integer, AvError> {

    private IAirVantageClient client;

    public LogoutTask(IAirVantageClient client) {
        this.client = client;
    }
    
    @Override
    protected AvError doInBackground(String... arg0) {
        AvError res = null;
        try {
            this.client.logout();
        } catch (IOException e) {
            res = new AvError("unexpected.error");
        } catch (AirVantageException e) {
            res = e.getError();
        }
        return res;
    }

}

File: mainActivity/src/main/java/com/sierrawireless/avphone/task/LogoutTask.kt
package com.sierrawireless.avphone.task

import java.io.IOException

import net.airvantage.model.AirVantageException
import net.airvantage.model.AvError
import net.airvantage.utils.IAirVantageClient
import android.os.AsyncTask

class LogoutTask internal constructor(private val client: IAirVantageClient) : AsyncTask<String, Int, AvError>() {

    override fun doInBackground(vararg arg0: String): AvError? {
        var res: AvError? = null
        try {
            this.client.logout()
        } catch (e: IOException) {
            res = AvError("unexpected.error")
        } catch (e: AirVantageException) {
            res = e.error
        }

        return res
    }

}
